docs: Remove all traces of "identity.pub" from docs

codablock opened the pull request:
"identity.pub" is referenced multiple times in CRDs and docs. This secret
is however never used in any place. Instead, the public key is derived from
the "identity" private key.

This commit/PR removes all traces of "identity.pub", including from older
api versions.

makkes reviewed (commented):
I'm fine with removing it here but we'd need to remove it from [flux/flux2](https://github.com/flux/flux2), too. The `identity.pub` field is included in 

* `create secret git`
* `create source git`
* `bootstrap git`

hiddeco commented:
> I'm fine with removing it here but we'd need to remove it from [flux/flux2](https://github.com/flux/flux2), too. 

That does not make sense to me, as we include it for sake of clarity when generating a private key for a user, and they e.g. need easy access to the public key later on (even if it can be derived from the PK).

makkes commented:
> > I'm fine with removing it here but we'd need to remove it from [fluxcd/flux2](https://github.com/fluxcd/flux2), too.
> 
> That does not make sense to me, as we include it for sake of clarity when generating a private key for a user, and they e.g. need easy access to the public key later on (even if it can be derived from the PK).

I kind of had the same thought. It's convenient to have the public in the secret for easy verification.

stefanprodan commented:
We've added the public key there, so people can extract it from the secret without needing openssh locally, just kubectl.

hiddeco commented:
I would say that this PR is still valid, because they are not required for the controller to function. However, the field should not be removed from the CLI commands.

hiddeco commented on api/v1beta1/gitrepository_types.go:
```diff
@@ -44,7 +44,7 @@ type GitRepositorySpec struct {
 	// The secret name containing the Git credentials.
 	// For HTTPS repositories the secret must contain username and password
 	// fields.
-	// For SSH repositories the secret must contain identity, identity.pub and
```
`v1beta1` has gone stale and does not require further modifications.

codablock replied:
Will remove the change. Does this also apply to v1alpha1?

hiddeco replied:
Yes, anything that isn't current.

hiddeco reviewed (approved):
This looks good to me from a documentation perspective. For your suggestions around reflecting it in the Status, I think further discussion is required.

hiddeco commented:
@codablock just discovered you likely changed `docs/api/source.md` by hand. It is generated through `make api-docs` however. Not sure why CI does not pick this up in the dirty Git state check, as running it locally gives me:

```console
$ git diff
diff --git i/docs/api/source.md w/docs/api/source.md
index c5cec6f..c068591 100644
--- i/docs/api/source.md
+++ w/docs/api/source.md
@@ -1630,8 +1630,8 @@ Artifact
 <td>
 <code>includedArtifacts</code><br>
 <em>
-<a href="#source.toolkit.fluxcd.io/v1beta2.*github.com/fluxcd/source-controller/api/v1beta2.Artifact">
-[]*github.com/fluxcd/source-controller/api/v1beta2.Artifact
+<a href="#source.toolkit.fluxcd.io/v1beta2.*./api/v1beta2.Artifact">
+[]*./api/v1beta2.Artifact
 </a>
 </em>
 </td>
```

makkes reviewed (changes requested):
Otherwise lgtm

codablock commented:
I've actually used `make api-docs` to update source.md. I just re-ran it on a vanilla main branch and the same change is performed. There seems to be something different on my machine compared to yours...not sure what it might be.

hiddeco commented:
@codablock what version of `gen-crd-api-reference-docs` do you have locally?

codablock commented:
@hiddeco 0.3.0, as downloaded by the Makefile. I also tried to downgrade to 0.2.0 just to ensure that it's not about the version, but that gave the same result. I've recently upgraded my golang installation to 1.18, maybe that's the reason. Unfortunately I don't have the time atm to verify if it would happen with 1.17 as well.

makkes commented:
Running `make api-docs` doesn't result in any changes when I run it on `main`, neither with Go 1.18 nor with 1.17.9. I wonder what's happening...

makkes commented:
btw, I filed an issue with gen-crd-api-reference-docs because the anchor and the text are wrong: https://github.com/ahmetb/gen-crd-api-reference-docs/issues/49.

Not related to this PR, I just stumbled upon the bug by reviewing the PR.

makkes commented:
@codablock feel free to cherry-pick https://github.com/makkes/source-controller/commit/929957d4b0f2105e6df2b2cf979ec962dc55a509 so you don't have to fix your setup in order to get this PR merged.

codablock commented:
@makkes Thanks. I though I reverted the changes already but somehow messed up the squash ¯\_(ツ)_/¯
I squashed your commit into mine, so now it should be fine.

makkes reviewed (approved):
lgtm

makkes commented:
@codablock please rebase your branch and force-push.
